lib/galaxy/jobs/handler.py

```diff
@@ -553,7 +553,14 @@ def __verify_job_ready(self, job, job_wrapper):
 
         if state == JOB_READY:
             state = self.__check_user_jobs(job, job_wrapper)
-        if state == JOB_READY and self.app.config.enable_quotas:
+        # If user has plugged a media, then they might have enough quota
+        # on their media; hence, we should not raise the "over quota" flag
+        # checking the default storage only. If their usage exceeds their
+        # total quota on all their media, ObjectStore raises appropriate
+        # exception(s).
+        if state == JOB_READY and self.app.config.enable_quotas and \
+                (job.user is not None and
+                 (job.user.active_storage_media is None or len(job.user.active_storage_media) == 0)):
```

Comment:
I'd make this a property on job.user - ``has_active_storage_media`` or something to clean up this piece of code. This whole chunk though should probably be refactored into a ``check_user_over_quota`` method somewhere...

Reply:
I implemented the property as you suggest. Reg. refactoring that chunk: I am leaving it as is for now for the sake of minimizing the changes with this PR, and we can modify as you suggested in subsequent PRs.